README.en

```diff
@@ -0,0 +1,357 @@
+approximately the same operation in Pire (for example, Scanner, SlowScanner и SimpleScanner).
+
+Each scanner has the State type that stores the state, and methods Initialize(State&)
+to return the initial machine state, Next(State&, char) to take the machine to the next state,
+Final(const State&) to report whether the state is valid. To check whether the string matches
+the pattern, it is required to run Initialize(), then Next() for each letter, then Final()
+to get the result.
+
+There is the Pire::Run(const Scanner&, Scanner::State&, const char* begin, const char* end)
+method that runs optimized Scanner::Next() for each character from the range.
+
+The pattern may include characters of the string start and end (^ and $ respectively).
+These characters are replaced with BeginMark and EndMark at compilation.
+
+To check the string with two scanners, you can use the Pire::Run(const Scanner1&, 
+const Scanner2&, Scanner1::State&, Scanner2::State&, const char* begin, const char* end)
+function. This function checks the string faster than two consecutive calls of
+the regular Run() function.
+
+Also, there is the RunHelper class that simplifies working with the Initialize-Step-Run-Step-Final
+cycle writing it as a single expression (see an example).
+
+Four more functions helpful in lexical analysis are LongestPrefix(), LongestSuffix(), ShortestPrefix(),
+and ShortestSuffix(). These functions returns the largest / the smallest prefix or suffix allowed by
+the pattern. Before LongestSuffix() or ShortestSuffix() call, it is required to run Fsm::Reverse().
+
+Machine union
+================
+
+As it is already mentioned above, the time to check whether the string matches the pattern
+does not depend on the pattern size. In terms of the checking speed it is better to use
+some complex expressions than a lot of simple ones. If it is required to check the text with
+a lot of simple expressions, it is possible to unite several scanners into one.
+
+The function Pire::Scanner::Glue() unites several machines. The resulting scanner
+checks whether the string matches all united patterns at a time. The Scanner::RegexpsCount()
+function returns common number of patterns. It is recommended to replace running Final() with
+AcceptedRegexps(const State&) that returns iterators with the [begin,end) semantics indicating the
+range of expression numbers to which the read string corresponds.
+
+Numbers start from 0 and, after union, numbers of expressions on the right are shifted
+by the expression numbers on the left.
+
+United machines can be united once again. However, it is not recommended to repeat union
+indefinitely because the number of states in the resulting scanner grows
+exponentially with each added pattern. The max size of the scanners can be specified in
+the maxSize parameter in the function Glue(). If the specified size is exceeded, an empty scanner
+will be returned (Size() == 0, Empty() == true).
```

Comment:
For the source paragraph

> Склеенные сканеры в свою очередь можно склеить между собой. В то же время процесс склейки
> нельзя повторять до бесконечности, потому как количество состояний в полученном автомате
> растёт экспоненциально с каждым добавленным паттерном. Иногда имеет смысл ограничить
> размер полученного автомата, указав ненулевой параметр maxSize в Glue(). В таком случае
> при превышении указанного размера возвращается пустой автомат (Size() == 0, Empty() == true).

https://translate.google.ru suggests

> The glued scanners, in turn, can be glued together. At the same time, the gluing process
> cannot be repeated indefinitely, because the number of states in the resulting machine
> grows exponentially with each added pattern. Sometimes it makes sense to limit
> the size of the resulting automaton by specifying a nonzero maxSize parameter in Glue (). In this case
> when the specified size is exceeded, an empty automaton is returned (Size () == 0, Empty () == true).

Which seams a bit clearer to me. First of all, it uses the words "to glue" and "the scanner" which are used for the same purpose in the code itself. This makes it easier to understand the code by reading the documentation. 
I would suggest using "to glue" and "glued" instead of "to unite" and "united" in previous paragraphs as well.
Even if it seams repetitive.